doc/tutorials/electrokinetics/electrokinetics.ipynb

```diff
@@ -4,385 +4,708 @@
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Up to this point there is no species present anywhere in the system and also no way for it to enter the system. Since the reaction is irreversible in our setup, we need to introduce some density of both the educt species to the system.\n",
+    "For that we setup two additional Dirichlet boundary conditions in the domain, where we fix the species density to a constant, non-zero value. This is done in such a way, that these two sources are spatially separated and only upon diffusion of the species through the channel, they will be able to react with each other. "
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "educt_species[0][BOX_L[1]//20,BOX_L[1]//4-1:BOX_L[1]//4+1,:].density_boundary = espressomd.electrokinetics.DensityBoundary(10.0)\n",
+    "educt_species[1][BOX_L[1]//20,3*(BOX_L[1]//4)-1:3*(BOX_L[1]//4)+1,:].density_boundary = espressomd.electrokinetics.DensityBoundary(10.0)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "With this, the system is now finally complete and we can start the integration. To see the system evolve, we will render a movie from the timeseries of the system."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import matplotlib.animation as animation\n",
+    "import matplotlib.quiver\n",
+    "import tempfile\n",
+    "import base64\n",
+    "\n",
+    "VIDEO_TAG = \"\"\"<video controls>\n",
+    " <source src=\"data:video/x-m4v;base64,{0}\" type=\"video/mp4\">\n",
+    " Your browser does not support the video tag.\n",
+    "</video>\"\"\"\n",
+    "\n",
+    "# set ignore 'divide' and 'invalid' errors\n",
+    "# these occur when plotting the flowfield containing a zero velocity\n",
+    "np.seterr(divide='ignore', invalid='ignore')\n",
+    "\n",
+    "def anim_to_html(anim):\n",
+    "    if not hasattr(anim, '_encoded_video'):\n",
+    "        with tempfile.NamedTemporaryFile(suffix='.mp4') as f:\n",
+    "            anim.save(f.name, fps=20, extra_args=['-vcodec', 'libx264'])\n",
+    "            with open(f.name, \"rb\") as g:\n",
+    "                video = g.read()\n",
+    "        anim._encoded_video = base64.b64encode(video).decode('ascii')\n",
+    "        plt.close(anim._fig)\n",
+    "    return VIDEO_TAG.format(anim._encoded_video)\n",
+    "\n",
+    "animation.Animation._repr_html_ = anim_to_html"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "box_width = lattice.shape[1]\n",
+    "box_height = lattice.shape[0]"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "scrolled": true
+   },
+   "outputs": [],
+   "source": [
+    "total_frames = 200\n",
```

Comment:
the video takes way too long to be produced. While the simulation is running, students can't use their interactive session to do subsequent tasks. Could the order be switched so this is last? It also seems like the most advanced system.